app/views/RoomView/index.js

```diff
@@ -681,6 +681,19 @@ class RoomView extends React.Component {
 		}
 	}
 
+	navToRoomPreview = async({ rid, name }) => {
+		const { user: { id: uid } } = this.props;
+		try {
+			const room = await RocketChat.canAccessRoom(rid, uid);
+			if (room) {
+				await Navigation.navigate('RoomsListView');
+				Navigation.navigate('RoomView', { rid, name, t: room.t });
+			}
+		} catch {
+			EventEmitter.emit(LISTENER, { message: I18n.t('error-not-allowed') });
```

Comment:
You're always telling the user he's not allowed to access the room.
What if he's without internet connection?

Reply:
Could I use something like [this example](https://reactnativeforyou.com/how-to-check-internet-connectivity-in-react-native-android-and-ios/) or there is already some code for this?